src/fsharp/TypeChecker.fs

```diff
@@ -13574,8 +13580,12 @@ module MutRecBindingChecking =
     /// Check a "module X = A.B.C" module abbreviation declaration
     let TcModuleAbbrevDecl (cenv:cenv) scopem env (id, p, m) = 
         let ad = env.eAccessRights
-        let mvvs = ForceRaise (ResolveLongIndentAsModuleOrNamespace cenv.tcSink ResultCollectionSettings.AllResults cenv.amap m OpenQualified env.eNameResEnv ad p false)
-        let modrefs = mvvs |> List.map p23 
+        let resolved =
+            match p with
+            | [] -> Result []
+            | id::rest -> ResolveLongIndentAsModuleOrNamespace cenv.tcSink ResultCollectionSettings.AllResults cenv.amap m true OpenQualified env.eNameResEnv ad id rest false
+        let mvvs = ForceRaise resolved
+        let modrefs = mvvs |> List.map p23
         if modrefs.Length > 0 && modrefs |> List.forall (fun modref -> modref.IsNamespace) then 
```

Comment:
These look good short-circuiting on empty, I wonder if a empty check on modrefs would improve the other side.